src/test/java/com/dagnis/restcountries/service/CountriesServiceTest.java

```diff
@@ -0,0 +1,134 @@
+        currencies.add(currency);
+        Country country = new Country("UK", "London", currencies, 900000, 1234);
+
+        List<Country> countries = new ArrayList<>();
+        countries.add(country);
+        countriesService.insertCountriesIntoDatabase(countries);
+
+        List<CurrencyEntity> currenciesFound = currencyRepository.findAll();
+        Assertions.assertEquals(currenciesFound.size(), 1);
+    }
+
+    @Test
+    void shouldBeFalseIfDataOlderThan24h() {
+        long DAY = 24 * 60 * 60 * 1000;
+
+        Currency currency = new Currency("GBP", "Pound", "G");
+        List<Currency> currencies = new ArrayList<>();
+        currencies.add(currency);
+        Country country = new Country("UK", "London", currencies, 900000, 1234);
+
+        List<Country> countries = new ArrayList<>();
+        countries.add(country);
+        countriesService.insertCountriesIntoDatabase(countries);
+
+        List<CountryEntity> countryEntities = countryRepository.findAll();
+        CountryEntity countryEntity = countryEntities.get(0);
+        countryEntity.setCreated(new Date(System.currentTimeMillis() - DAY * 2));
+        countryRepository.save(countryEntity);
+
+        Assertions.assertFalse(countriesService.checkIfDataNotStale());
+    }
+
+    @Test
+    void shouldBeTrueIfDataNewerThan24h() {
+
+        Currency currency = new Currency("GBP", "Pound", "G");
+        List<Currency> currencies = new ArrayList<>();
+        currencies.add(currency);
+        Country country = new Country("UK", "London", currencies, 900000, 1234);
+
+        List<Country> countries = new ArrayList<>();
+        countries.add(country);
+        countriesService.insertCountriesIntoDatabase(countries);
+
+        Assertions.assertTrue(countriesService.checkIfDataNotStale());
+    }
+
+    @Test
+    void shouldNotThrowOnDeletingFromDatabase() {
+        Currency currency = new Currency("EUR", "Euro", "E");
+        List<Currency> currencies = new ArrayList<>();
+        currencies.add(currency);
+        Country country = new Country("Austria", "Vienna", currencies, 900000, 1234);
+
+        List<Country> countries = new ArrayList<>();
+        countries.add(country);
+        countriesService.insertCountriesIntoDatabase(countries);
+
+        countryRepository.deleteAll();
+        currencyRepository.deleteAll();
+    }
+
+    @Test
+    void countryShouldHaveCurrency() {
+        Currency currency = new Currency("EUR", "Euro", "E");
+        List<Currency> currencies = new ArrayList<>();
+        currencies.add(currency);
+        Country country = new Country("Austria", "Vienna", currencies, 900000, 1234);
+
+        List<Country> countries = new ArrayList<>();
+        countries.add(country);
+        countriesService.insertCountriesIntoDatabase(countries);
+
+        Optional<CountryEntity> countryEntity = Optional.ofNullable(countryRepository.findAll().get(0));
+        Assertions.assertNotNull(countryEntity.get().getCurrencies());
```

Comment:
check for size, or a match on fields
a list may be not null, but empty if it wasn't saved accidentally